src/librustc_codegen_ironox/x86_asm_printer.rs

```diff
@@ -0,0 +1,79 @@
+use super::ModuleIronOx;
+
+use context::CodegenCx;
+use value::{Value, Instruction};
+
+use rustc::mir::mono::Stats;
+use rustc::util::nodemap::FxHashMap;
+use rustc_codegen_ssa::traits::MiscMethods;
+use std::cell::{Cell, RefCell};
+
+/// A convenience macro for appending instructions to a string.
+macro_rules! asm {
+    (
+        $m:ident,
+        $(
+            $fmt:expr; [ $($args:expr),* ]
+        ),*
+    ) => {
+        let mut asm_lines = vec![$(format!($fmt, $($args),*)),*];
```

Comment:
I'm not sure it's a big deal, but I wondered if you wanted to try and avoid creating this intermediate vector which isn't doing much except slow things down? I guess you can do something like:

```rust
for s in $(args) {
   $m.push_str(s);
}
```

Of course, I can't even vaguely remember Rust's macro syntax, so that might not be possible...

Reply:
I'll remove the macros for now, but I will address this in my next PR>